inventory-upgrade.py

```diff
@@ -0,0 +1,350 @@
+                    return True
+        return False
+
+    def backup(self, backup_suffix):
+        """Backup the inventory"""
+        backup_path = self.path.with_suffix(backup_suffix)
+        if backup_path.exists():
+            suffix = backup_suffix + "-" + str(int(round(datetime.now().timestamp())))
+            backup_path = self.path.with_suffix(suffix)
+            print(
+                f"Warning: {self.path.with_suffix(backup_suffix)} already exists.\n"
+                f"The inventory will be backup up into: {backup_path}\n"
+            )
+        shutil.copytree(self.path, backup_path)
+
+    def write_inventory(self, backup=True, backup_suffix=".backup", delete_legacy=True):
+        if self.is_legacy:
+            raise ValueError(
+                "This script can't write legacy inventory. Please upgrade it first."
+            )
+        if backup:
+            self.backup(backup_suffix)
+
+        # Segregate the mandatory variables from the custom ones, there are is the template.
+        # The custom variables are written at the end of the file
+        required_variables = {}
+        grouped_variables = []
+        previous_prefix = None
+        for key, value in self.content.items():
+            if key in REQUIRED_VARS:
+                required_variables[key] = value
+            else:
+                prefix = key.split("_")[0]
+                # Same prefix, add to same dict
+                if previous_prefix == prefix:
+                    grouped_variables[-1]["content"][key] = value
+                else:
+                    grouped_variables.append(
+                        {"prefix": prefix, "content": {key: value}}
+                    )
+                previous_prefix = prefix
+
+        template = Template(VARS_TEMPLATE_DEFINITION)
+        with self.vars_path.open(mode="w+") as file:
+            file.write(template.render(required_variables))
+            for group in grouped_variables:
+                if len(group["content"]) > 1:
+                    file.write(f"\n## {group['prefix']}\n")
+                else:
+                    file.write(f"\n")
+                yaml.dump(
+                    group["content"],
+                    file,
+                    indent=2,
+                    allow_unicode=True,
+                    default_flow_style=False,
+                    sort_keys=False,
+                    Dumper=MyDumper,
+                    width=160,
+                )
+
+        if delete_legacy:
+            for path in self.legacy_vars_paths:
+                if path.exists():
+                    path.unlink()
+
+
+class MyDumper(yaml.Dumper):
+    """configure better indentation and list & stuff like that
+    Ref: https://stackoverflow.com/questions/25108581/python-yaml-dump-bad-indentation/39681672#39681672
+    """
+
+    def increase_indent(self, flow=False, indentless=False):
+        return super(MyDumper, self).increase_indent(flow, False)
```

Comment:
```suggestion
        return super().increase_indent(flow, False)
```
No more needed since python3